lib/apiservers/engine/backends/convert/state.go

```diff
@@ -78,7 +78,7 @@ func State(info *models.ContainerInfo) *types.ContainerState {
 				// most likely the process was killed via the cli
 				// or received a sigkill
 				exitCode = 137
-			} else if info.ProcessConfig.Status == "" && exitCode == 0 {
+			} else if info.ProcessConfig.Status == "" && exitCode == -1 {
```

Comment:
yup - the only reason we were seeing `0` in master is because we're overwriting the exit code from the container in this path.